src/DynamicData/Cache/Internal/ChangeSetMergeTracker.cs

```diff
@@ -48,6 +48,30 @@ public void RemoveItems(IEnumerable<KeyValuePair<TKey, TObject>> items, IObserve
         EmitChanges(observer);
     }
 
+    public void RefreshItems(IEnumerable<TKey> keys, IObserver<IChangeSet<TObject, TKey>> observer)
+    {
+        var sourceCaches = _selectCaches().ToArray();
+
+        // Update the Published Value for each item being removed
+        if (keys is IList<TKey> list)
+        {
+            // zero allocation enumerator
```

Comment:
I included the zero allocation enumerator a long time ago after I realised that implementing an IChangeSet allocated when it is iterated.

Later I re-implemented the change sets to inherit List<T> rather than to delegate it.  This hacky way of doing things is a throw back to the earlier way of doing things,  Ideally we should remove IChangeSet altogether but in practice, it would break every person on this earth who has included a custom Dynamic Data extension.

At an even later point, I wondered whether there is a single person in this universe that ever implemented a custom IChangeSet and thought it to be unlikely.  To test this hypothesis I introduced and internal direct cast to List<T>, see [CacheChangeSetEx.cs](https://github.com/reactivemarbles/DynamicData/blob/main/src/DynamicData/Cache/CacheChangeSetEx.cs) and use it internally in several key places.

This was in 2020 and there has not been a single complaint about any code being broken by this cast.  I think we can safely just cast to the concrete type throughout the code base.  Just to be thorough, I think we should use the ToConcreteType() extension but change the implementation to this:

```cs
    public static ChangeSet<TObject, TKey> ToConcreteType<TObject, TKey>(this IChangeSet<TObject, TKey> changeSet)
        where TObject : notnull
        where TKey : notnull
    {
        if (changeSet is ChangeSet<TObject, TKey> concreteList)
        {
            return concreteList;
        }

        throw new NotSupportedException("Dynamic Data does not support a custom implementation of IChangeSet");
    }
```

Reply:
I am using the `ToConcreteType` extension when enumerating the ChangeSets.  I'm not sure if I'm using the thing for `EnumerableIList.Create` correctly because I just cloned from an example.

I'm happy to change the implementation of `ToConcreteType` but it should probably be another PR because it will affect so many things not related to this PR, and if it needs to be rolled back, it's easier if it's isolated.